### PR TITLE
cstor-main-container containerization

### PR DESCRIPTION
**Pull request by gkGaneshR**
1. Why is this change necessary ?
  The docker container for cstor main container needs to be created

2. How does this change address the issue ?
   The corresponding dockerfile and travis yaml are uploaded

3. How to verify this change ?
   docker pull openebs/cstor-main-container:ci

Signed-off-by: gkGaneshR <gkganesh126@gmail.com>

### CODE REVIEW

File: cstor-container [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+cd ../zfs
+sh autogen.sh
+
+if [ $ZFS_BUILD_TAGS = 0 ]; then
+   ./configure --enable-code-coverage=yes --enable-debug --enable-uzfs=yes;
+    make;
+else
+   ./configure --enable-code-coverage=yes --enable-debug;
+    make --no-print-directory -s pkg-utils pkg-kmod;
+    sudo dpkg -i *.deb;
```
Comment (gila) [inline cstor-container]: Lets not do that... pkg install should be manual

File: cstor-container [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+fi
+
+cd ../zfs
+sh autogen.sh
+
+if [ $ZFS_BUILD_TAGS = 0 ]; then
+   ./configure --enable-code-coverage=yes --enable-debug --enable-uzfs=yes;
+    make;
+else
+   ./configure --enable-code-coverage=yes --enable-debug;
```
Comment (gila) [inline cstor-container]: in both cases we are building kernel modules as well, thats not needed

Comment (gila) [reply cstor-container]: --with-config=user should suffice here

File: cstor-container [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+cd spl
+git checkout spl-0.7-release
+sh autogen.sh
+./configure
+
+if [ $ZFS_BUILD_TAGS = 0 ]; then
+   make;
+else
+   make --no-print-directory -s pkg-utils pkg-kmod;
+   sudo dpkg -i *.deb;
```
Comment (gila) [inline cstor-container]: don't install pkgs magically it can hose systems

File: cstor-container [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+git clone https://github.com/openebs/zfs
+#git checkout zfs-0.7-release
+#cd zfs
+#bash -c "pushd ."
+#cd /usr/src/gtest
+#sudo cmake CMakeLists.txt
+#sudo make
+#sudo cp *.a /usr/lib
+#bash -c "popd"
```
Comment (gila) [inline cstor-container]: why are these lines commented out? can they be removed?

File: cstor-prerequisites [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+set -e
+
+#sudo apt-get -qq update
+  sudo apt-get install --yes -qq build-essential autoconf libtool gawk alien fakeroot linux-headers-$(uname -r) libaio-dev
+  sudo apt-get install --yes -qq zlib1g-dev uuid-dev libattr1-dev libblkid-dev libselinux-dev libudev-dev libssl-dev
+  sudo apt-get install --yes -qq lcov
+#packages for tests
+  sudo apt-get install --yes -qq parted lsscsi ksh attr acl nfs-kernel-server fio
+  sudo apt-get install --yes -qq libgtest-dev cmake
+
```
Comment (gila) [inline cstor-prerequisites]: So this only works for debian systems...

File: cstor-container (hunk 2 above)
Comment (gila) [reply cstor-container]: why are we doing debug builds?

File: Dockerfile.maincontainer [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+
+ARG BUILD_DATE
+LABEL org.label-schema.name="cstor"
+LABEL org.label-schema.description="OpenEBS cstor"
+LABEL org.label-schema.url="http://www.openebs.io/"
+LABEL org.label-schema.vcs-url="https://github.com/openebs/cstor"
+LABEL org.label-schema.schema-version="1.0"
+LABEL org.label-schema.build-date=$BUILD_DATE
+
+ENTRYPOINT entrypoint.sh "${CSTOR_NETWORK}"
```
Comment (gila) [inline Dockerfile.maincontainer]: whats the purpose of this entrypoint?

File: cstor-prerequisites (hunk 5 above)
Comment (gkGaneshR) [reply cstor-prerequisites]: @gila I got this from cloudbyte repo only. Rgiht now, the container image is in ubuntu. In that case, is the code tested for other distributions?

Comment (gila) [reply cstor-prerequisites]: Well, we have a couple of distro's we use today (i.e with JIVA) but we certainly should support fedora/centos next to ubuntu for sure.

File: Dockerfile.maincontainer (hunk 6 above)
Comment (gkGaneshR) [reply Dockerfile.maincontainer]: for starting tgt(zrepl), ssh, rsyslog

File: cstor-container (hunk 4 above)
Comment (gkGaneshR) [reply cstor-container]: will remove them,

File: cstor-container (hunk 2 above)
Comment (gkGaneshR) [reply cstor-container]: A)./configure --enable-code-coverage=yes  --enable-uzfs=yes;
make;
B)A)./configure --enable-code-coverage=yes  --with-config=user --enable-uzfs=yes;
make;
Did you mean like this?  Which one shall we use, A or B? If B, how is user set? is that env var? or just the string- user?

File: cstor-container (hunk 3 above)
Comment (gkGaneshR) [reply cstor-container]: I got this dpkg from cloudbytestorage travis yaml file. I'll remove them

File: Dockerfile.maincontainer (hunk 6 above)
Comment (gila) [reply Dockerfile.maincontainer]: but why do we need to restart SSH? and why is SSH in that container to begin with?

Comment (gkGaneshR) [reply Dockerfile.maincontainer]: @vishnuitta Can you pls confirm if we need to have ssh, since you had asked me to keep before?

**Comment by gkGaneshR** [comment]
Will close this and raise new pr coz., of some travis build issue.

**Pull request closed without merging**